File: eureka/ios/nativeLib/nativeLib/Models/Responses/CommonResponse.swift
//
//  CommonResponse.swift
//  eureka
//
//  Created by Harshada Memane on 31/03/21.
//

import Foundation


class CommonResponse: ResponseBase {
    
  private var _userId: String = ""
  private var _authId: String = ""
    
  internal override init(pResponseBase: ResponseBase) {
    super.init(pResponseBase:pResponseBase)
  }
  
  internal override init(pErrorCode: ResultCodeEnum) {
    super.init(pErrorCode: pErrorCode)
  }
    
  internal override init(pErrorCode: ResultCodeEnum, pMessage: String) {
    super.init(pErrorCode: pErrorCode, pMessage: pMessage)
  }
    
  internal override init(pSimuId: String, pSimuMessage: String) {
    super.init(pSimuId: pSimuId, pSimuMessage: pSimuMessage)
  }
    
  internal override func getResponseStr() -> String {
    do {
      var mResultMap: [String:Any] = [:]
      var result: [String:Any] = getResponseMap()
      var mData: [String:String] = [:]
        
      mData["userId"] = _userId
      mData["AuthenticationId"] = _authId
        
      result["data"] = mData as NSObject
        
      mResultMap["result"] = result as NSObject
        
      let jsonData = try JSONSerialization.data(withJSONObject: mResultMap, options: [])
      let jsonString = String(data: jsonData, encoding: String.Encoding.ascii)!
      return jsonString
    } catch _ {
      return ""
    }
  }
    
  internal func getUserId() -> String {
    return _userId
  }
    
  internal func setUserId(pUserId: String) {
    _userId = pUserId
  }
    
  internal func getAuthId() -> String {
    return _authId
  }
    
  internal func setAuthId(pAuthId: String) {
    _authId = pAuthId
  }
}

File: eureka/android/library/src/main/java/com/lifeplus/Pojo/Responses/CommonResponse.java
package com.lifeplus.Pojo.Responses;

import com.google.gson.Gson;
import com.lifeplus.Pojo.Enum.ResultCodeEnum;

import java.util.HashMap;

public class CommonResponse extends ResponseBase {
    private String _userId = "";
    private String _authId = "";

    public CommonResponse(ResultCodeEnum pErrorCode) {
        super(pErrorCode);
    }

    public CommonResponse(ResultCodeEnum pErrorCode, String pMessage) {
        super(pErrorCode, pMessage);
    }

/*
    public CalibrationResponse(String pMessage) {
        super(pMessage);
    }
*/

    public CommonResponse(String pSimuId, String pSimuMessage) {
        super(pSimuId, pSimuMessage);
    }

    @Override
    public String getResponseStr() {
        HashMap<String, Object> mResultMap = new HashMap<>();

        HashMap<String, Object> result = getResponseMap();
        HashMap<String, String> mData = new HashMap<>();
        mData.put( "userId", _userId );
        mData.put( "AuthenticationId", _authId );
        result.put( "data", mData );

        mResultMap.put( "result",  result);
        return new Gson().toJson( mResultMap );
    }

    public String getUserId() {
        return _userId;
    }

    public void setUserId(String pUserId) {
        _userId = pUserId;
    }

    public String getAuthId() {
        return _authId;
    }

    public void setAuthId(String pAuthId) {
        _authId = pAuthId;
    }
}